packages/aws-cdk/lib/api/bootstrap/bootstrap-template.yaml

```diff
@@ -500,6 +520,66 @@ Resources:
             - - Fn::Sub: "arn:${AWS::Partition}:iam::aws:policy/AdministratorAccess"
       RoleName:
         Fn::Sub: cdk-${Qualifier}-cfn-exec-role-${AWS::AccountId}-${AWS::Region}
+      PermissionsBoundary:
+        Fn::If:
+          - DefaultPermissionsBoundary
+          - Ref: PermissionsBoundaryPolicy
+          - Fn::If:
+              - CustomPermissionsBoundary
+              - Fn::Sub: 'arn:${AWS::Partition}:iam:${AWS::Region}:${AWS::AccountId}:policy/${InputPermissionsBoundary}'
+              - Ref: AWS::NoValue
+  PermissionsBoundaryPolicy:
+    Condition: DefaultPermissionsBoundary
```

Comment:
One thing I just thought of, should we just create the policy regardless and only _attach_ based on
the parameter? I'm not sure there is a downside to creating the policy.

It might actually be an issue to conditionally create it. I don't think you can delete a policy if
it is attached to any roles, so if the permissions boundary has been attached to any roles you can't
re-bootstrap without the policy.